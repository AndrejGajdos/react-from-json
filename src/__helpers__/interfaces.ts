export interface Components {
  Bun: object[];
  Patty: object[];
  PattyIngredient: object[];
}

export interface Mapping {
  Bun: React.ReactNode;
  Burger: React.ReactNode;
  Patty: React.ReactNode;
  PattyIngredient: React.ReactNode;
}
