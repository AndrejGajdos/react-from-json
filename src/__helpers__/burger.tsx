import * as React from "react";

export const Bun = ({ variant }: { variant: string }) => (
  <div>{`${variant} bun`}</div>
);

export const Patty = ({
  id,
  size,
  ingredient
}: {
  id: number;
  size: string;
  ingredient: React.ReactNode;
}) => (
  <div id={`${id}`}>
    {`${size} patty`}
    {ingredient}
  </div>
);

export const PattyIngredient = ({ variant }: { variant: string }) => (
  <div>{variant}</div>
);

export const Burger = ({
  bun,
  cheese,
  children
}: {
  bun: React.ReactNode;
  chain: string;
  cheese: boolean;
  children: React.ReactNode;
}) => (
  <div>
    <div>{bun}</div>
    <div>{cheese && "cheese"}</div>
    <div>{children}</div>
    <div>{bun}</div>
  </div>
);
