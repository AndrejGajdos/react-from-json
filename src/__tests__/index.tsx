import * as React from "react";
import ReactFromJSON from "../";
import * as renderer from "react-test-renderer";

import { Mapping, Components } from "../__helpers__/interfaces";
import {
  mapping,
  recursiveEntry,
  flatEntry,
  flatComponents
} from "../__helpers__/data";

class BurgerReactFromJSON extends ReactFromJSON<Mapping, Components> {
  render(): JSX.Element {
    return super.render();
  }
}

describe("ReactFromJSON", () => {
  it("to render a recursive entry", () => {
    const tree = renderer
      .create(<BurgerReactFromJSON entry={recursiveEntry} mapping={mapping} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("to render a flat entry with a components attribute", () => {
    const tree = renderer
      .create(
        <BurgerReactFromJSON
          entry={flatEntry}
          mapping={mapping}
          components={flatComponents}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
