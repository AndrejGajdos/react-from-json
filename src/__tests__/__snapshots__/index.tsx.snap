// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactFromJSON to render a flat entry with a components attribute 1`] = `
<div>
  <div>
    <div>
      sesame bun
    </div>
  </div>
  <div>
    cheese
  </div>
  <div>
    <div
      id="0"
    >
      large patty
      <div>
        impossible
      </div>
    </div>
    <div
      id="1"
    >
      large patty
      <div>
        beef
      </div>
    </div>
  </div>
  <div>
    <div>
      sesame bun
    </div>
  </div>
</div>
`;

exports[`ReactFromJSON to render a recursive entry 1`] = `
<div>
  <div>
    <div>
      sesame bun
    </div>
  </div>
  <div>
    cheese
  </div>
  <div>
    <div
      id="undefined"
    >
      large patty
      <div>
        impossible
      </div>
    </div>
    <div
      id="undefined"
    >
      large patty
      <div>
        beef
      </div>
    </div>
  </div>
  <div>
    <div>
      sesame bun
    </div>
  </div>
</div>
`;
